presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcess.java

```diff
@@ -0,0 +1,198 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution;
+
+import com.facebook.airlift.http.client.HttpClient;
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.Session;
+import com.facebook.presto.SystemSessionProperties;
+import com.facebook.presto.client.ServerInfo;
+import com.facebook.presto.execution.TaskId;
+import com.facebook.presto.execution.TaskInfo;
+import com.facebook.presto.execution.TaskManagerConfig;
+import com.facebook.presto.execution.TaskSource;
+import com.facebook.presto.execution.scheduler.TableWriteInfo;
+import com.facebook.presto.server.TaskUpdateRequest;
+import com.facebook.presto.spark.execution.http.PrestoSparkHttpWorkerClient;
+import com.facebook.presto.spark.execution.property.NativeExecutionConnectorConfig;
+import com.facebook.presto.spark.execution.property.NativeExecutionNodeConfig;
+import com.facebook.presto.spark.execution.property.NativeExecutionSystemConfig;
+import com.facebook.presto.spark.execution.property.WorkerProperty;
+import com.facebook.presto.sql.planner.PlanFragment;
+import io.airlift.units.Duration;
+import org.apache.spark.SparkFiles;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.URI;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class NativeExecutionProcess
+{
+    private static final Logger log = Logger.get(NativeExecutionProcess.class);
+    private static final int NATIVE_PROCESS_DEFAULT_PORT = 7777;
+    private static final String WORKER_CONFIG_FILE = "/config.properties";
+    private static final String WORKER_NODE_CONFIG_FILE = "/node.properties";
+    private static final String WORKER_CONNECTOR_CONFIG_FILE = "/catalog/";
+    private static final Duration MAX_ERROR_DURATION = new Duration(2, TimeUnit.MINUTES);
+    private Process process;
+    private final Session session;
+    private final PrestoSparkHttpWorkerClient workerClient;
+    private final URI location;
+    private final int port;
+    private final TaskManagerConfig taskManagerConfig;
+    private final NativeExecutionSystemConfig systemConfig;
+    private final NativeExecutionNodeConfig nodeConfig;
+    private final NativeExecutionConnectorConfig connectorConfig;
+    private NativeExecutionTask nativeExecutionTask;
```

Comment:
Why do we put this here? It does not seem like been used.

Reply:
it will be used by caller (e.g native operator) by calling `public NativeExecutionTask getTask()`

Reply:
This field can be marked as final.